### ISSUE TITLE
Proposal: Update Go template to include an example that works out-of-the-box

### ISSUE THREAD

**Issue by salaboy** [knative-sandbox/kn-plugin-func#839]
Following the Spring Boot template, which includes a working function and cloud events routing based on CE-Type or an expression, it will be great if the Go template includes some working examples following the same lines. We can apply the same example to other languages too. 

The Go Template just include an empty Handle function, that by default uses a signature that doesn't return a CloudEvent after processing the request. 

I will create a PR with a simple example following the Spring Boot template example.

**Comment by zroubalik** [comment]
I think this is being tracked here: https://github.com/knative-sandbox/kn-plugin-func/issues/572

**Comment by salaboy** [comment]
Is it ok to keep this as a sub-issue from #572 so we can use it to discuss the proposed example for Go only?

**Comment by salaboy** [comment]
Added a PR with a simple example.. follows the same approach as the `spring-boot` template.. it helps to quickly have a runnable function to test and demonstrate

**Comment by lkingland** [comment]
The builtin Functions are not examples.  They are the minimum necessary to echo its imput (for integration testing), and pre-populate the user's new Function with the default function signature, in the language chosen.

For Example Functions, see the [Function Examples](https://github.com/boson-project/templates) templates repository.

**Comment by lkingland** [comment]
> The Go Template just include an empty Handle function, that by default uses a signature that doesn't return a CloudEvent after processing the request.

Please see the discussion in #606 for the reasoning behind this decision.

tl;dr returning an error is go idiomatic, and avoids an extra import.  However, the default Function should have no actual implementation other than to echo back the event received, so perhaps we can use the more complex function signature and simply return the inbound `event`.

**Comment by salaboy** [comment]
@lkingland ok this is cool.. so I've created this repo to test: 

https://github.com/salaboy/func-templates

Do you think if we create a repo `func-templates` or `func-examples` into the `knative-sandbox` org we can document the examples in the official docs?

**Comment by lkingland** [comment]
> @lkingland ok this is cool.. so I've created this repo to test:
> 
> https://github.com/salaboy/func-templates
> 
> Do you think if we create a repo `func-templates` or `func-examples` into the `knative-sandbox` org we can document the examples in the official docs?

@salaboy I think so, yes.  In fact that is exactly one of the use-cases in mind when adding the "extensible template repositories" feature!
